### ISSUE TITLE
Incorrect Builder.from generated with javac when overriding methods

### ISSUE THREAD

**Issue by tjwilson90** [immutables/immutables#437]
For example
```
public interface Foo {
    String getFoo();
    List<String> getFoos();
}

@Value.Immutable
public interface Bar extends Foo {
    @Override List<String> getFoos();
}
```

When generated with ecj, the `ImmutableBar.Builder.from(Object)` method looks like
```
    private void from(Object object) {
      long bits = 0;
      if (object instanceof Bar) {
        Bar instance = (Bar) object;
        if ((bits & 0x1L) == 0) {
          addAllFoos(instance.getFoos());
          bits |= 0x1L;
        }
      }
      if (object instanceof Foo) {
        Foo instance = (Foo) object;
        if ((bits & 0x1L) == 0) {
          addAllFoos(instance.getFoos());
          bits |= 0x1L;
        }
        foo(instance.getFoo());
      }
    }
```

When generated with javac, it looks like
```
    private void from(Object object) {
      if (object instanceof Bar) {
        Bar instance = (Bar) object;
        addAllFoos(instance.getFoos());
      }
      if (object instanceof Foo) {
        Foo instance = (Foo) object;
        foo(instance.getFoo());
      }
    }
```

These have different behavior when `object` is a `Foo`, but not a `Bar`.

This appears to be a regression added between 2.2.10 and 2.2.11.

**Comment by elucash** [comment]
Thank you for revealing this!

**Comment by elucash** [comment]
please, can you try with newly released 2.3, it should be fixed now. thank you!

**Comment by tjwilson90** [comment]
Yes, this appears to have fixed the issue, thanks.
